src/operator/subgraph/dnnl/dnnl_identity_property.h

```diff
@@ -0,0 +1,187 @@
+      in_status = InStatus::kStart;
+      matched_list_.clear();
+      matched_list_.emplace_back(&seed_node);
+      return true;
+    }
+    return false;
+  }
+
+  bool SelectInput(const BiDirectedNode& n, const BiDirectedNode& input_node) override {
+    if (in_status == InStatus::kFail || in_status == InStatus::kSuccess ||
+        input_node.node->is_variable()) {
+      return false;
+    }
+
+    switch (in_status) {
+      case InStatus::kStart:
+        if (input_node.node->op()) {
+          in_status = InStatus::kSuccess;
+          matched_list_.emplace_back(&input_node);
+          return true;
+        }
+      default:
+        return false;
+    }
+    return false;
+  }
+
+  bool SelectOutput(const BiDirectedNode& n, const BiDirectedNode& output_node) override {
+    return false;
+  }
+
+  std::vector<BiDirectedNode*> Filter(const std::vector<BiDirectedNode*>& candidates) override {
+    if (in_status == InStatus::kFail || in_status != InStatus::kSuccess) {
+      return std::vector<BiDirectedNode*>(0);
+    } else {
+      std::vector<BiDirectedNode*> ret;
+      for (auto i : matched_list_) {
+        auto non_const_i = const_cast<BiDirectedNode*>(i);
+        if (std::find(candidates.begin(), candidates.end(), non_const_i) != candidates.end()) {
+          ret.push_back(non_const_i);
+        }
+      }
+      return ret;
+    }
+  }
+
+  void Reset() override {
+    CHECK_GE(matched_list_.size(), 1);
+    auto new_selector = SgDNNLIdentitySelector();
+    new_selector.Select(*matched_list_[0], nullptr);
+    *this = new_selector;
+  }
+};
+
+inline bool IsIdentityNode(const nnvm::ObjectPtr node) {
+  return node->op() == Op::Get("_npi_copy") || node->op() == Op::Get("Dropout");
+}
+
+class SgDNNLIdentityProperty : public SubgraphProperty {
+ public:
+  SgDNNLIdentityProperty() {}
+
+  static SubgraphPropertyPtr Create() {
+    static const std::string& name = "DNNL Identity optimization pass";
+    auto property                  = std::make_shared<SgDNNLIdentityProperty>();
+    property->SetAttr<std::string>("property_name", name);
+    property->SetAttr<bool>("inference_only", true);
+    return property;
+  }
+
+  nnvm::ObjectPtr CreateSubgraphNode(const nnvm::Symbol& sym,
+                                     const int subgraph_id = 0) const override {
+    nnvm::NodeEntry identity_node_entry;
+    for (auto nentry : sym.outputs) {
+      if (nentry.node->op() && IsIdentityNode(nentry.node)) {
+        identity_node_entry = nentry;
+      }
+    }
+
+    auto last_node = identity_node_entry.node;
+    nnvm::Symbol new_sym;
+    new_sym.outputs.emplace_back(last_node);
+
+    nnvm::ObjectPtr org_node;
+    DFSVisit(new_sym.outputs, [&](const nnvm::ObjectPtr& node) {
+      if (!IsIdentityNode(node) && node->op()) {
+        org_node = node;
+      }
+    });
+
+    // Create copy of original node
+    nnvm::ObjectPtr n = nnvm::Node::Create();
+    n->attrs          = org_node->attrs;
+    CHECK(n->attrs.op);
```

Comment:
```suggestion
    CHECK(n->op());
```
To make it clear that this check refers to the `n->op()` used below.

Reply:
Done